src/modules/sync/advancedDiff.ts

```diff
@@ -0,0 +1,529 @@
+import { ContentTypeElements } from "@kontent-ai/management-sdk";
+import chalk from "chalk";
+import { readFileSync, writeFileSync } from "fs";
+import open from "open";
+import { resolve } from "path";
+
+import { SyncParams } from "../../commands/diff/diff.js";
+import { logError, logInfo } from "../../log.js";
+import {
+  isContentTypeData,
+  isContentTypeSnippetData,
+  isCountLimitation,
+  isDefaultValue,
+  isDependsOn,
+  isExternalIdReference,
+  isExternalIdReferenceArray,
+  isMaximumTextLength,
+  isObjectReference,
+  isObjectReferenceArray,
+  isTaxonomyRequestModel,
+  isValidationRegex,
+} from "../../utils/typeguards.js";
+import { RequiredCodename } from "../../utils/types.js";
+import { DiffModel, DiffObject } from "./types/diffModel.js";
+import { PatchOperation, ReplacePatchOperationValue } from "./types/patchOperation.js";
+
+type DiffData = DiffModel & SyncParams;
+type EntityPathHandler = {
+  regex: RegExp;
+  entity: (() => string) | ((match: string[]) => string);
+};
+
+export const writeDiffToFile = (diffData: DiffData) => {
+  const template = readFileSync(resolve("./", "diffTemplate.html"), "utf-8");
+  const resolvedTemplate = template.replace(
+    /{{.*?}}/g,
+    (match) => templateHandlerMap.get(match)?.(diffData) ?? match,
+  );
+  const outputPath = resolve(
+    diffData.outPath!,
```

Comment:
outPath is optional only in the scope of diff itself. for advanced logging, it's required (flag `-a` will not work without `-o` and vice versa, as they imply each other in yargs. I'm fine adding a check but I thought this was a valid use case for non-null assertion. what do you say?

Reply:
on the other hand, I see one may include `-o` parameter without value. however, that will just generate the diff in the root folder (or not, since it might be in some protected folder if used with `npx`). will wait for what you have to say :)

Reply:
I understand your logic, however, we will be exposing to functions so that they can be used programmatically inside other scripts. Then Yargs logic won't help us to determine whether it's in correct state. Moreover, as I see it we are also trying to make the functions as much decoupled from other logic as possible. For these reasons, I still think it would be better to be handled somehow.

Reply:
added check for output path presence